statgen/locuszoom
# Datasources for grch38

# Purpose
LZ.js should offer support for grch38 builds. This adds new datasources, as well as global config options to control genome build consistently via a single parameter across all requests.

# How to test this
An LZ plot requires 4 core pieces of data. New API services allow us to support each of them.
- Assoc
- LD
- Genes
- Recomb

Compare old and new demo pages for differences. The LD data should largely be a drop in replacement. The following index.html region requires 2 pages of LD data, so verify that all points are colored (pages fill from left to right side of plot):
`index.html?chrom=10&start=114306616&end=114823842`

Genes track may show more or slightly different gene annotations after the version upgrade. Existing deployments often specify a source explicitly, and so they will continue to use the old gencode dataset version.

If the current portal public datasets do not provide build 38 association studies, I can further evaluate via using locuszoom-tabix locally.

# TODO
- [x] Standardize mechanisms for querying build across all new and existing data sources
- [x] Add support for dynamically selecting which LD population or build to render
- [x] Update LD source based on API feedback/ discussion with Daniel
- [x] Evaluate impact of changing default gene source / build, incl wrt ExAC "constraint" info.
- [x] Upgrade PheWAS source and make build param consistent
